gp2gp-translator/src/main/resources/templates/nackMessageTemplate.mustache

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!-- The Application Acknowledgement message MCCI_IN010000UK13, is defined in the HL7v3 MIM, which can be found here:
+https://data.developer.nhs.uk/dms/mim/4.2.00/Index.htm-->
+
+<MCCI_IN010000UK13 xmlns="urn:hl7-org:v3"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="urn:hl7-org:v3 urn:data.developer.nhs.uk/dms/mim/4.2.00/Schemas/MCCI_IN010000UK13.xsd"
+                   type="Message"
+>
+    <id root="{{messageId}}"/>
+    <creationTime value="{{creationTime}}"/>
+    <versionCode code="V3NPfIT4.2.00"/>
+    <interactionId root="2.16.840.1.113883.2.1.3.2.4.12" extension="MCCI_IN010000UK13"/>
+    <processingCode code="P"/>
+    <processingModeCode code="T"/>
+    <acceptAckCode code="NE"/>
+    <acknowledgement typeCode="AE">
+        <messageRef>
+            <id root="{{messageRef}}"/>
+        </messageRef>
+    </acknowledgement>
+    <communicationFunctionRcv typeCode="RCV" type="CommunicationFunction">
+        <device type="Device" classCode="DEV" determinerCode="INSTANCE">
+            <id root="1.2.826.0.1285.0.2.0.107" extension="{{toAsid}}"/>
+        </device>
+    </communicationFunctionRcv>
+    <communicationFunctionSnd typeCode="SND" type="CommunicationFunction">
+        <device type="Device" classCode="SND">
+            <id root="1.2.826.0.1285.0.2.0.107" extension="{{fromAsid}}"/>
+        </device>
+    </communicationFunctionSnd>
+    <ControlActEvent classCode="CACT" moodCode="EVN" type="ControlAct">
+        <author1 typeCode="AUT" type="Participation">
+            <AgentSystemSDS classCode="AGNT" type="RoleHeir">
+                <agentSystemSDS classCode="DEV" determinerCode="INSTANCE" type="Device">
+                    <id root="1.2.826.0.1285.0.2.0.107" extension="{{fromAsid}}"/>
+                </agentSystemSDS>
+            </AgentSystemSDS>
+        </author1>
+        <reason typeCode="RSON">
```

Comment:
Since this is the same type of message as the ackMessage and we're using Mustache templates, you could leave only one template. Check if `nackCode` is not null and include `justifyingDetectedIssueEvent` element based on that check.